Add deprecated exercises article (#107)
exercism/docs: building/tracks/deprecated-exercises.md

# Deprecated exercises

> Deleting an exercise is a destructive action it would delete all users' solutions to that exercise.

Reviewer: ```suggestion
Deleting an exercise is a destructive action as it would delete all users' solutions to that exercise.
```

> Therefore exercises can never be deleted.

Reviewer: ```suggestion
Therefore, exercises can **never** be deleted.
```

Instead, we allow exercises to be deprecated.

Deprecating an exercise has the following behaviours:

> - Users who have downloaded or started the exercise can access their existing solutions as normal

Reviewer: ```suggestion
- Users who have downloaded or started the exercise can access their existing solution as normal
```

> - Users show have **not** downloaded or started it, will not see or start the exercise.

Reviewer: ```suggestion
- Users who have **not** downloaded or started the exercise, will not see or be able to start it.
```

To help with this, we have config rules that accompany a deprecated exercise

> Deprecated exercises must have the following set in `config.json`:

Reviewer: ```suggestion
Deprecated exercises must have the following properties set in `config.json`:
```

- **`uuid`** (unchanged from when it was active)

> - **`slug`**

Reviewer: I'd probably also add the unchanged note here too, as renaming the slug doesn't make much sense (while still technically possible)

```suggestion
- **`slug`** (unchanged from when it was active)
```

- **`status`** set to `deprecated`

The following should be empty:

- `prerequisites`
- `practices`
- `concepts`

> All other fields in deprecated exercises can be safely changed or removed in accordance with the wider config rules.

Reviewer: Could you also update the `linting.md` document?
